https-request.js

```diff
@@ -54,7 +54,27 @@ module.exports = async (context, hostname, method, requestPath, body, headers) =
                         reject(`Invalid JSON: ${json}`)
                     }
                 })
+            }
+
+            const req = https.request(options, res => {
+                if (res.statusCode === 301 || res.statusCode === 302) {
+                    const url = res.headers.location
+                    const match = url.match(/^https:\/\/([^/]+)(.*)$/)
+                    if (!match) throw new Error(`Unsupported redirect URL: ${url}`)
+
+                    const options = {
+                        port: 443,
+                        hostname: match[1],
```

Comment:
We could theoretically encounter a redirect url with an explicit port, which would then be part of `match[1]`, but that's unlikely to cause us any trouble in practice.

Reply:
I like that solution.